src/main/java/io/nats/client/Options.java

```diff
@@ -225,8 +229,16 @@
     /**
      * Property used to configure a builder from a Properties object. {@value}, see
      * {@link Builder#dataPortType(String) dataPortType}.
+     * 
+     * Replaced with {@link Builder#natsChannelFactory(NatsChannelFactory) natsChannelFactory}.
      */
+    @Deprecated
     public static final String PROP_DATA_PORT_TYPE = PFX + "dataport.type";
+    /**
+     * Property used to configure a builder from a Properties object. {@value}, see
+     * {@link Builder#natsChannelFactory(NatsChannelFactory) natsChannelFactory}.
+     */
+    public static final String PROP_NATS_CHANNEL_FACTORY = PFX + "channel.factory";

```

Comment:
Shouldn't the factory be fixed and it uses some other property to figure out what it generates? Maybe an example of what this value is